session.go

```diff
@@ -239,35 +241,6 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 			}
 			return
 
-		// handle allocation request
-		case l := <-s.allocateHandle:
-			// Check if link name already exists, if so then an error should be returned
-			if linksByKey[l.Key] != nil {
-				l.err = fmt.Errorf("link with name '%v' already exists", l.Key.name)
-				l.RX <- nil
-				continue
-			}
-
-			next, ok := handles.Next()
-			if !ok {
-				// handle numbers are zero-based, report the actual count
-				l.err = fmt.Errorf("reached session handle max (%d)", s.handleMax+1)
-				l.RX <- nil
-				continue
-			}
-
-			l.Handle = next       // allocate handle to the link
-			linksByKey[l.Key] = l // add to mapping
-			l.RX <- nil           // send nil on channel to indicate allocation complete
```

Comment:
Just checking - we don't need this <-nil anymore because we block on the `send` channel at the caller, correct?

Reply:
For reference.

https://github.com/Azure/go-amqp/blob/main/link.go#L242-L259

The way this used to work is that `link.attach()` would send its `*l` to the `s.allocateHandle` channel.  It would then wait on `l.RX` which signaled that `*l` was ready for use.  Since we now directly allocate the handle, we don't need to perform this handshake.